src/core/net/ip6.cpp

```diff
@@ -1068,6 +1069,22 @@ Error Ip6::SendRaw(OwnedPtr<Message> aMessagePtr)
 
     if (header.GetDestination().IsMulticast())
     {
+#if OPENTHREAD_CONFIG_FILTER_LOCAL_ORIGINATED_MULTICAST_PACKETS
+        // For multicast packets originated from this device, only forward to thread
+        // network if destination is in the multicast listener table
+        if (!aMessagePtr->IsOriginThreadNetif() && Get<ThreadNetif>().HasUnicastAddress(header.GetSource()))
```

Comment:
```
// For multicast packets originated from this device, only forward to thread
// network if destination is in the multicast listener table
```

Maybe better to clarify  this comment to make it clear that it applies to speicifc messages originating from host and passed to OT stack (in `SendRaw()`).

For example, MLE advertisements originate from the device and they are multicast and should definitely be forwarded to Thread network. Another example, MeshCop Net Diag query to a multicast address.

Reply:
Updated the comment, thanks